experimental/components/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! 🚧 `icu_experimental` is the development crate of the `ICU4X` project.
+//!
+//! It will usually undergo a major SemVer bump for every ICU4X release. Components in this
+//! crate will eventually stabilize and move to the `icu` crate.
+
+// https://github.com/unicode-org/icu4x/blob/main/docs/process/boilerplate.md#library-annotations
+#![cfg_attr(not(any(test, feature = "std")), no_std)]
+// No boilerplate, each module has their own
+#![allow(clippy::module_inception)]
+
+extern crate alloc;
+
+pub mod compactdecimal;
+pub mod dimension;
+pub mod displaynames;
+pub mod personnames;
+pub mod relativetime;
+pub mod transliterate;
+pub mod unicodeset_parse;
+
+#[doc(hidden)]
```

Comment:
Nit: We don't normally put the provider module in doc(hidden), though we could consider making them internal when we implement #4467

Reply:
Well we have n provider modules in this crate, but `Baked` needs to be in `crate::provider` because the constructor macro expects that. We can reexport `Baked` in foo::provider to have it public?

Reply:
I think it's fine for `crate::provider` to behave as it normally does with baked data, and for the data structs to be defined in `crate::foo::provider`.